### PR TITLE
Bug bash - Fix bugs in multi GPU benchmarks

### PR DESCRIPTION
**Pull request by abuccts**
__Description__

Fix bugs in multi GPU benchmarks.

__Major Revisions__

- Add sb deploy content
- Fix inline if-expression syntax
- Fix quote escape issue in bash command
- Add custom env in config
- Update default config for multi GPU
- Fix wrong parameters in torch distributed microbenchmarks
- Update MANIFEST.in to include jinja2 template
- Require jinja2 minimum version
- Fix occasional duplicate output in Ansible runner
- Fix mixed color issue in log

### CODE REVIEW

File: tests/executor/test_executor.py [hunk 1]
```diff
@@ -68,8 +68,8 @@ def test_get_arguments(self):
             ), expected_matmul_args
         )
         expected_bert_models_args = \
-            '--duration 0 --num_warmup 16 --num_steps 128 --batch_size 16 ' \
-            '--precision float32 float16 --model_action train inference'
+            '--duration 0 --num_warmup 16 --num_steps 128 --batch_size 8 ' \
+            '--precision float32 float16 --model_action train'
```
Comment (cp5555) [inline tests/executor/test_executor.py:72]: Why not test inference?

File: superbench/config/default.yaml [hunk 2]
```diff
@@ -4,27 +4,41 @@ superbench:
   benchmarks:
     kernel-launch:
       enable: true
+      modes: &local_mode_config
+        - name: local
+          proc_num: 8
+          prefix: CUDA_VISIBLE_DEVICES={proc_rank}
+          parallel: yes
```
Comment (cp5555) [inline superbench/config/default.yaml:11]: Make it in a general place.

File: superbench/config/default.yaml [hunk 3]
```diff
@@ -4,27 +4,41 @@ superbench:
...
         - pytorch
-    gpt_models:
+    sharding-matmul:
       enable: true
-      modes:
+      modes: &pytorch_mode_config
         - name: torch.distributed
           proc_num: 8
-          node_num: all
+          node_num: 1
```
Comment (cp5555) [inline superbench/config/default.yaml:31]: Make it in a general place.

File: superbench/config/default.yaml (hunk 2 above)
Comment (abuccts) [reply superbench/config/default.yaml:11]: updated

File: superbench/config/default.yaml (hunk 3 above)
Comment (abuccts) [reply superbench/config/default.yaml:17]: updated

File: superbench/runner/runner.py [hunk 4]
```diff
@@ -113,7 +115,7 @@ def __get_mode_command(self, benchmark_name, mode):
                 torch_distributed_suffix=(
                     'superbench.benchmarks.{name}.parameters.distributed_impl=ddp '
                     'superbench.benchmarks.{name}.parameters.distributed_backend=nccl'
-                ).format(name=benchmark_name),
+                ).format(name=benchmark_name) if benchmark_name.endswith('_models') else '',
```
Comment (cp5555) [inline superbench/runner/runner.py:118]: remove this.

File: tests/runner/test_runner.py [hunk 5]
```diff
@@ -106,9 +106,9 @@ def test_get_mode_command(self):
                     '--use_env --no_python --nproc_per_node=8 '
                     '--nnodes=1 --node_rank=$NODE_RANK '
                     '--master_addr=$MASTER_ADDR --master_port=$MASTER_PORT '
-                    'sb exec -c sb.config.yaml -C superbench.enable=foo '
-                    'superbench.benchmarks.foo.parameters.distributed_impl=ddp '
-                    'superbench.benchmarks.foo.parameters.distributed_backend=nccl'
+                    'sb exec -c sb.config.yaml -C superbench.enable=foo_models '
+                    'superbench.benchmarks.foo_models.parameters.distributed_impl=ddp '
+                    'superbench.benchmarks.foo_models.parameters.distributed_backend=nccl'
```
Comment (cp5555) [inline tests/runner/test_runner.py:111]: Rollback.

**Review by cp5555 — APPROVED** [review]

**Review by guoshzhao — APPROVED** [review]

**Pull request merged**